certbot-dns-dnsimple/local-oldest-requirements.txt

```diff
@@ -1,2 +1,3 @@
-acme[dev]==0.21.1
-certbot[dev]==0.21.1
+acme[dev]==0.25.0
+-e .[dev]
+dns-lexicon==2.2.1
```

Comment:
I don't think this works unfortunately. The idea behind these `local-oldest-requirements.txt` files were to list the "oldest" dependencies on packages found "locally" in this repo. When used this new way, it appears the constraints in `tools/dev_constraints.txt` overrides the requirements here. See https://travis-ci.org/certbot/certbot/jobs/477426491#L1341.

While I think we should use this version of Lexicon in the oldest tests for all of our DNS plugins where we are claiming to support 2.2.1, we can't do it everywhere. For instance, the OVH plugin requires `dns-lexicon>=2.7.14`.

We haven't had this problem yet where we're testing different pieces of Certbot with different oldest constraints. Do you want to propose a solution for this? This could (and maybe should) be done in a different PR, however, if we want to delay fixing the problem, I think we should put `dns-lexicon==2.7.14` in `tools/oldest_constraints` in a new section for Ubuntu 18.04 Bionic to ensure we keep compatibility with Lexicon 2.x.

Reply:
Ah, I did not caught that. Indeed `local-oldest-requirements.txt` is of low priority not necessarily fulfilled, and local oldest requirement is not designed to isolate specific requirements for each project. I will think of a more general solution, it is not the first time I hit its limitations.

In the meantime, yes, I think putting a little higher `dns-lexicon` is a correct temporary solution.

Reply:
Oldest requirement updated.

Reply:
Great! Can you make an issue to track this work on separate oldest requirements?

Reply:
PR is on its way ^^